File: Vjezbe_3/particle.py
import matplotlib.pyplot as plt
import math

class Particle:
    def init(self,v0,theta,x0,y0,dt):
        self.v0 = v0
        self.theta = theta
        self.xi = x0
        self.yi = y0
        self.dt = dt
        self.x = []
        self.y = []
        self.x.append(self.xi)
        self.y.append(self.yi)
        self.kut = self.theta*math.pi/180
        self.vx = self.v0*math.cos(self.kut)
        self.vy = self.v0*math.sin(self.kut)

    def reset(self):
        self.v0 = 0
        self.theta = 0
        self.xi = 0
        self.yi = 0
        self.dt = 0
        self.x = []
        self.y = []
        self.kut = 0
        self.v0x = 0
        self.v0y = 0

    def __move(self):
        a = 9.81
        self.xi = self.xi + self.vx*self.dt
        self.vy = self.vy - a*self.dt
        self.yi = self.yi + self.vy*self.dt
        self.x.append(self.xi)
        self.y.append(self.yi)

    def range(self):
        while self.yi >= 0:
            self.__move()
        return max(self.x)

    def analiticki_domet(self):
        D = ((self.v0**2)*math.sin(2*self.kut))/9.81
        return D

    def plot_trajectory(self):
        plt.figure("Graf za trenutno stanje")
        plt.plot(self.x,self.y)
        plt.title("x-y graf")
        plt.xlabel("x[m]")
        plt.ylabel("y[m]")
        plt.show()

File: Vjezbe_7/particle.cpp
#define _USE_MATH_DEFINES
#include <math.h>
#include "particle.h"

Particle::Particle(double v0, double theta, double x0, double y0, double step) //constructor
    {
        ti = 0;
        dt = step;
        xi = x0;
        yi = y0;
        kut = theta*M_PI/180;
        vx = v0*cos(kut);
        vy = v0*sin(kut);
    }

void Particle::evolve()
    {
        while (yi >= 0) {
            vx += 0.;
            vy += g*dt;

            xi += vx*dt;
            yi += vy*dt;

            ti += dt;
        }
    }

double Particle::range()
    {
        evolve();
        D = xi;
        return D;
    }

double Particle::time()
    {
        evolve();
        t = ti;
        return t;
    }